### PR TITLE
Fix(Script/Quest) A Cry For Help

### LINKED ISSUE #5368: [Quest] A Cry For Help - Cowlen's reaction is not scripted
Originally reported https://github.com/chromiecraft/chromiecraft/issues/446

#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [x] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

When reaching the destination of the escort quest A Cry For Help (https://www.wowhead.com/quest=9528/a-cry-for-help) only the daughter - Magwin - is scripted.  Her father Cowlen sits there doing nothing.

https://user-images.githubusercontent.com/81375770/115316159-188de800-a179-11eb-83e6-b50c431b4234.mp4

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Cowlen also says one line and gets up when Magwin reaches him. 
https://www.wowhead.com/quest=9528/a-cry-for-help#comments:id=662675

Full video of the quest from TBC classic is also online, timestamp pointing to the missing event
https://youtu.be/PILIuPCUz40?t=372

See also the quotes for Cowlen available in the wowgaming database:
https://wowgaming.altervista.org/aowow/?npc=17311

##### STEPS TO REPRODUCE THE PROBLEM:

1. Accept the quest from Magwin (https://wowgaming.altervista.org/aowow/?quest=9528)
2. Complete the quest
3. Notice that Cowlen at the end just sits there.

##### EXTRA NOTES:

Type here.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/b8b5f68f98ff8771a6ec87ea6805ceced91c406a

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/97909879-quest-a-cry-for-help-cowlen-s-reaction-is-not-scripted?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Branel**
## Changes Proposed:
Add missing dialogue from Cowlen.
Timers and emotes.

## Issues Addressed:

- Closes #5368 
- Closes https://github.com/chromiecraft/chromiecraft/issues/445

## SOURCE:

https://github.com/chromiecraft/chromiecraft/issues/446

Huge thanks to @Helias for finding the original fix.
All credit goes to @SnapperRy for his original work on this issue.

## Tests Performed:

Tested on my local windows 10 server

## How to Test the Changes:

1. Be draenei, .level 10
2. .go c id 17312
3. do the quest

## Known Issues and TODO List:

I need to redo my test on a fresh Azeroth Core as I have configured the SmartAI a lot while trying to fix it myself, will update on more info soon.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 1]
```diff
@@ -245,23 +255,63 @@ class npc_magwin : public CreatureScript
...
+        {
+            _events.Update(diff);
+
+            if (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_ACCEPT_QUEST:
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                            Talk(SAY_START, player);
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:285]: Add brackets here and for the other if's aswell please.

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 2]
```diff
@@ -245,23 +255,63 @@ class npc_magwin : public CreatureScript
...
+                        me->setFaction(FACTION_QUEST);
+                        _events.ScheduleEvent(EVENT_START_ESCORT, 1000);
+                        break;
+                    case EVENT_START_ESCORT:
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                            npc_escortAI::Start(true, false, player->GetGUID());
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:291]: +

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 3]
```diff
@@ -245,23 +255,63 @@ class npc_magwin : public CreatureScript
...
+                        _events.ScheduleEvent(EVENT_STAND, 2000);
+                        break;
+                    case EVENT_STAND: // Remove kneel standstate. Using a separate delayed event because it causes unwanted delay before starting waypoint movement.
+                        me->SetByteValue(UNIT_FIELD_BYTES_1, 0, 0);
+                        break;
+                    case EVENT_TALK_END:
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                            Talk(SAY_END1, player);
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:299]: +

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 4]
```diff
@@ -245,23 +255,63 @@ class npc_magwin : public CreatureScript
...
+                        _events.ScheduleEvent(EVENT_COWLEN_TALK, 2000);
+                        break;
+                    case EVENT_COWLEN_TALK:
+                        if (Creature* cowlen = me->FindNearestCreature(NPC_COWLEN, 50.0f, true))
+                            cowlen->AI()->Talk(SAY_COWLEN);
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:304]: +

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 5]
```diff
@@ -265,7 +265,9 @@ class npc_magwin : public CreatureScript
                         break;
                     case 29:
                         if (Creature* cowlen = me->FindNearestCreature(NPC_COWLEN, 50.0f, true))
+                            {
                             Talk(EMOTE_HUG, cowlen);
+                            }
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp]:
```suggestion
                        {
                            Talk(EMOTE_HUG, cowlen);
                        }
```

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 6]
```diff
@@ -282,26 +284,34 @@ class npc_magwin : public CreatureScript
                 {
                     case EVENT_ACCEPT_QUEST:
                         if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                            {
                             Talk(SAY_START, player);
+                            }
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp]:
```suggestion
                        {
                            Talk(SAY_START, player);
                        }
```

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 7]
```diff
@@ -282,26 +284,34 @@ class npc_magwin : public CreatureScript
...
                         me->setFaction(FACTION_QUEST);
                         _events.ScheduleEvent(EVENT_START_ESCORT, 1000);
                         break;
                     case EVENT_START_ESCORT:
                         if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                            {
                             npc_escortAI::Start(true, false, player->GetGUID());
+                            }
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp]:
```suggestion
                        {
                            npc_escortAI::Start(true, false, player->GetGUID());
                        }
```

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 8]
```diff
@@ -282,26 +284,34 @@ class npc_magwin : public CreatureScript
...
                         _events.ScheduleEvent(EVENT_STAND, 2000);
                         break;
                     case EVENT_STAND: // Remove kneel standstate. Using a separate delayed event because it causes unwanted delay before starting waypoint movement.
                         me->SetByteValue(UNIT_FIELD_BYTES_1, 0, 0);
                         break;
                     case EVENT_TALK_END:
                         if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                            {
                             Talk(SAY_END1, player);
+                            }
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp]:
```suggestion
                        {
                            Talk(SAY_END1, player);
                        }
```

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 9]
```diff
@@ -282,26 +284,34 @@ class npc_magwin : public CreatureScript
...
                         _events.ScheduleEvent(EVENT_COWLEN_TALK, 2000);
                         break;
                     case EVENT_COWLEN_TALK:
                         if (Creature* cowlen = me->FindNearestCreature(NPC_COWLEN, 50.0f, true))
+                            {
                             cowlen->AI()->Talk(SAY_COWLEN);
+                            }
```
Comment (Hacki95) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp]:
```suggestion
                        {
                            cowlen->AI()->Talk(SAY_COWLEN);
                        }
```

**Review by Hacki95 — CHANGES_REQUESTED** [review]
Correct indentation.

**Comment by Branel** [comment]
Thank you for the feedback Hacki :D

File: data/sql/updates/pending_db_world/rev_1620469549972464900.sql [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+UPDATE `creature_text` SET `emote`=5 WHERE `creatureid`=17312 AND `groupid`=2;
+UPDATE `creature_text` SET `emote`=22 WHERE `creatureid`=17312 AND `groupid`=3;
+UPDATE `creature_text` SET `emote`=5 WHERE `creatureid`=17312 AND `groupid`=4;
+
+DELETE FROM `creature_text` WHERE `creatureid`=17311;
+INSERT INTO `creature_text` (`creatureid`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(17311, 0, 0, (SELECT `MaleText` FROM `broadcast_text` WHERE `ID`=13749), 12, 7, 100, 0, 0, 0, 13749, 0, "Cowlen");
+
+UPDATE `script_waypoint` SET `waittime`=3000 WHERE `entry`=17312 AND `pointid`=28;
+UPDATE `script_waypoint` SET `waittime`=5000 WHERE `entry`=17312 AND `pointid`=29;
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1620469549972464900.sql:12]: missing new line at the end of file

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 11]
```diff
@@ -245,23 +255,70 @@ class npc_magwin : public CreatureScript
...
+                        break;
+                    case EVENT_START_ESCORT:
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, _player))
+                        {
+                            npc_escortAI::Start(true, false, player->GetGUID());
+                        }
+                        _events.ScheduleEvent(EVENT_STAND, 2000);
+                        break;
+                    case EVENT_STAND: // Remove kneel standstate. Using a separate delayed event because it causes unwanted delay before starting waypoint movement.
+                        me->SetByteValue(UNIT_FIELD_BYTES_1, 0, 0);
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:300]:
```suggestion
                        me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_STAND_STATE, UNIT_STAND_STATE_STAND);
```

**Pull request merged**